File: CDAC Project/payment-dotnet/PaymentService/Models/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PaymentService.Models
{
    [Table("payments")]
    public class Payment
    {
        [Key]
        [Column("payment_id")]
        public long PaymentId { get; set; }

        [Column("sale_id")]
        public long SaleId { get; set; }

        [Column("amount")]
        public double Amount { get; set; }

        [Column("status")]
        public string Status { get; set; }

        [Column("gateway_order_id")]
        public string GatewayOrderId { get; set; }

        [Column("gateway_payment_id")]
        public string GatewayPaymentId { get; set; }

        [Column("gateway_signature")] // Added for completeness, usually mapped
        public string? GatewaySignature { get; set; } // Matches Java entity

        [Column("paid_at")]
        public DateTime? PaidAt { get; set; }
    }
}

File: CDAC Project/FV/FVBS/src/main/java/com/farmer_vendor/entity/Payment.java
package com.farmer_vendor.entity;

import java.time.LocalDateTime;

import jakarta.persistence.*;

@Entity
@Table(name = "payments")
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long paymentId;

    @OneToOne
    @JoinColumn(name = "sale_id", nullable = false)
    private Sale sale;

    private Double amount;

    private String paymentMethod;

    // INITIATED / SUCCESS / FAILED
    private String status;

    // Razorpay IDs
    @Column(nullable = false, unique = true)
    private String gatewayOrderId;     // razorpay_order_id

    private String gatewayPaymentId;   // razorpay_payment_id

    @Column(length = 512)
    private String gatewaySignature;   // razorpay_signature

    private LocalDateTime paidAt;

	public Long getPaymentId() {
		return paymentId;
	}

	public void setPaymentId(Long paymentId) {
		this.paymentId = paymentId;
	}

	public Sale getSale() {
		return sale;
	}

	public void setSale(Sale sale) {
		this.sale = sale;
	}

	public Double getAmount() {
		return amount;
	}

	public void setAmount(Double amount) {
		this.amount = amount;
	}

	public String getPaymentMethod() {
		return paymentMethod;
	}

	public void setPaymentMethod(String paymentMethod) {
		this.paymentMethod = paymentMethod;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

	public String getGatewayOrderId() {
		return gatewayOrderId;
	}

	public void setGatewayOrderId(String gatewayOrderId) {
		this.gatewayOrderId = gatewayOrderId;
	}

	public String getGatewayPaymentId() {
		return gatewayPaymentId;
	}

	public void setGatewayPaymentId(String gatewayPaymentId) {
		this.gatewayPaymentId = gatewayPaymentId;
	}

	public String getGatewaySignature() {
		return gatewaySignature;
	}

	public void setGatewaySignature(String gatewaySignature) {
		this.gatewaySignature = gatewaySignature;
	}

	public LocalDateTime getPaidAt() {
		return paidAt;
	}

	public void setPaidAt(LocalDateTime paidAt) {
		this.paidAt = paidAt;
	}

    // getters & setters
    
    
}
